vue/components/ui/molecules/bezier-curve/bezier-curve.vue

```diff
@@ -0,0 +1,377 @@
+            v-on:mousedown="onMouseDown"
+        />
+        <circle
+            class="point"
+            v-bind:cx="bezier.cx1"
+            v-bind:cy="bezier.cy1"
+            v-show="showPoints"
+            id="controlpoint-1"
+            v-on:touchstart="onTouchStart"
+            v-on:mousedown="onMouseDown"
+        />
+        <circle
+            class="point"
+            v-bind:cx="bezier.cx2"
+            v-bind:cy="bezier.cy2"
+            v-show="showPoints"
+            id="controlpoint-2"
+            v-on:touchstart="onTouchStart"
+            v-on:mousedown="onMouseDown"
+        />
+    </svg>
+</template>
+
+<style lang="scss" scoped>
+@import "css/variables.scss";
+
+.bezier-curve.disabled {
+    pointer-events: none;
+}
+
+.bezier-curve .curve {
+    fill: none;
+    stroke: #000000;
+    stroke-width: 5px;
+}
+
+.bezier-curve .slope {
+    stroke: #008080;
+    stroke-width: 1;
+}
+
+.bezier-curve .point {
+    cursor: pointer;
+    fill: #ffffff;
+    filter: drop-shadow(2px 2px 1px #00000066);
+    r: 0.3rem;
+    stroke: #000000;
+    stroke-width: 0.5px;
+}
+
+body.tablet .bezier-curve .point,
+body.mobile .bezier-curve .point {
+    filter: drop-shadow(5px 5px 4px #0000005e);
+    r: 2rem;
+    stroke-width: 1px;
+}
+
+.bezier-curve .point:hover {
+    cursor: pointer;
+    filter: drop-shadow(0px 0px 3px #007bff);
+}
+
+.bezier-curve .point:active {
+    cursor: move;
+}
+</style>
+
+<script>
+export const BezierCurve = {
+    name: "bezier-curve",
+    props: {
+        /**
+         * Bezier curve endpoint 1 X coordinate.
+         */
+        x1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 1 Y coordinate.
+         */
+        y1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 2 X coordinate.
+         */
+        x2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve endpoint 2 Y coordinate.
+         */
+        y2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 X coordinate.
+         */
+        cx1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 Y coordinate.
+         */
+        cy1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 X coordinate.
+         */
+        cx2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 Y coordinate.
+         */
+        cy2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * The number of pixels of the svg viewport
+         * in the format "`width`x`height`".
+         */
+        resolution: {

```

Comment:
Separate into width and height.

Reply:
I think that could be misleading as width and height wouldn't increase the width or height of the component
I would suggest `resolutionWidth` and `resolutionHeight`